### ISSUE TITLE
Rewrite in Swift

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/macos-wallpaper#12]
Would make it easier to maintain.

Pull request welcome :)

**Comment by Shurlow** [comment]
Hey! Whats the status on this issue/project?

Made an initial pass at a swift re-wite here: https://github.com/Shurlow/macos-wallpaper

**Comment by sindresorhus** [comment]
> Made an initial pass at a swift re-write here: Shurlow/macos-wallpaper

Looks like a great start :) 

Don't forget this part: https://github.com/sindresorhus/macos-wallpaper/blob/8594438a8b06d88f6afb027274f7049b33f0b14d/wallpaper.m#L64-L95

> Are you accepting PR's at the moment? Id be interested in contributing 😄

Yes, definitely. A pull request would be awesome.

**Comment by tom-james-watson** [comment]
Hey guys,

Great work on this lib! If you are working on a rewrite, can you please consider adding support for setting on multiple spaces?

Check out how I've done it on https://github.com/tom-james-watson/wallpaper-macos, using https://github.com/tom-james-watson/macos-space-change. 

Thanks!

**Comment by sindresorhus** [comment]
@Shurlow Still interested in doing a PR? No worries if not though. Just thought I'd ask.

**Comment by Shurlow** [comment]
Yeah still in the works. Had a bit of trouble picking a good swift command line package. Ended up using this one: https://github.com/jakeheis/SwiftCLI. I couldn't find one that fully supports the current syntax unfortunately.

What are your thoughts on changing the syntax to use `set` and `get` methods? I'd rather not make breaking changes, but this is the closest/cleanest i've been able to come up with.

See: https://github.com/Shurlow/macos-wallpaper#usage

**Comment by sindresorhus** [comment]
> What are your thoughts on changing the syntax to use set and get methods? I'd rather not make breaking changes, but this is the closest/cleanest i've been able to come up with.

I'm ok with that. It might even be better as it's more explicit.
